### PR TITLE
New fly.toml template for V2 apps

### PR DESCRIPTION
**Pull request by dangra**
New apps won't get its starting application config from Web, instead they will leverage on http_service and top level checks sections

```toml
# Generated by:
#   fly launch --no-deploy --image nginx --force-machines --name machines-default -o personal -r ord

app = "machines-default"
primary_region = "ord"

[build]
  image = "nginx"

[http_service]
  internal_port = 8080
  force_https = true

[checks]
  [checks.alive]
    type = "tcp"
    interval = "15s"
    timeout = "2s"
    grace_period = "5s"
```   

The fly.toml generated from Web looked like:
```toml
# Generated by:
#   fly launch --no-deploy --image nginx --force-nomad --name nomad-default -o personal -r ord

app = "nomad-default"
kill_signal = "SIGINT"
kill_timeout = 5
primary_region = "ord"
processes = []

[build]
  image = "nginx"

[env]

[experimental]
  auto_rollback = true

[[services]]
  http_checks = []
  internal_port = 8080
  processes = ["app"]
  protocol = "tcp"
  script_checks = []
  [services.concurrency]
    hard_limit = 25
    soft_limit = 20
    type = "connections"

  [[services.ports]]
    force_https = true
    handlers = ["http"]
    port = 80

  [[services.ports]]
    handlers = ["tls", "http"]
    port = 443

  [[services.tcp_checks]]
    grace_period = "1s"
    interval = "15s"
    restart_limit = 0
    timeout = "2s"
```

Some other things in this PR:
* Lot of preflight tests for different `fly launch` flags
* Preflight tests for cases when fly.toml exists and/or app exists
* Test --force-nomad and --force-machines edge cases when app exists and its platform is already set
* **Do not rescan if [build] section exists**
* Scanners will modify HttpService and top level checks for v2 apps
* This will also respect --region and toml's primary_region if found, otherwise will ask the user

### CODE REVIEW

File: test/preflight/fly_launch_test.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/superfly/flyctl/test/preflight/testlib"
+)
+
+// TODO: list of things to test
+// - sourceInfo clone vs launch modes
+// - sourceInfo remote dockerfile vs local
+// END
```
Comment (alichay) [inline test/preflight/fly_launch_test.go:16]: still relevant?

**Review by alichay — APPROVED** [review]
you've been doing some great modernizing! one seemingly unfinished TODO that I felt I should flag, but it can probably ship anyway

File: test/preflight/fly_launch_test.go (hunk 1 above)
Comment (dangra) [reply test/preflight/fly_launch_test.go:16]: Ohh. those are things done by scanning code that I'd like to test not that I changed them. 

There are lot more flags and even frameworks I'd like to add flighttesting for but had to draw a line to ship this changes.
